### ISSUE TITLE
Differences between code and paper

### ISSUE THREAD

**Issue by sheep-for** [wenhuchen/KGPT#5]
Nice job! I have several questions as follow:
(1) I found some differences between code and paper, e.g., property embedding hasn't been used, and various parameter values (batch size/learning rate) are not the same in different experimental setup (training/challenge/a few shots). I wonder whether the SOTA released in the original paper can be implemented by following the settings in the scripts.
(2) I run the sequence-encoder-few-shot-1% experiment at WEGNLG dataset. Only the batch size is modified to 16 due to insufficient resources, and the epoch becomes 4000 correspondingly for the same total steps. I achieve 19.9 BLEU (paper: 24.7). Is there any important information or tricks i miss?
(3) The proposed model has achieved significant performance under zero-shot and few-shot settings. Could you show more details about how many training instances could obtain a competitive results against other baselines (Fully-Supervised Results)?
Thanks a lot!

**Comment by wenhuchen** [comment]
Hi, thanks for your interest.
1) property embedding doesn't help too much (the performance is almost the same), I simplify the code to eliminate that part. It's very easy to add this part back. The batch size/learning rate for all the different experiments needs to be tuned, the paper only reports the full-shot setting.
2) first of all, the results are all on the test-set using the https://github.com/wenhuchen/Data-to-text-Evaluation-Metric to evaluate, not based on NLTK (dev set). The NLTK dev set results are lower than using the official E2E-NLG BLEU script. Did you use the validation set to select the highest performing checkpoint? The learning rate might not be the optimum, if you still cannot reproduce, try to adjust it a little bit.
3) I can't remember this result, I remember WebNLG is roughly 10-20%?

**Comment by sheep-for** [comment]
Thanks for your prompt reply.
For the second question, I do use the official E2E-NLG BLEU script as well as the validation set to select the highest performing checkpoint. Does it mean that batch size and learning rate have a markable impact on the performance among all parameters? I'll do more try later.

**Comment by wenhuchen** [comment]
Could you run multiple times to see the variance? I'm not sure whether that's the case. It seems that the gap is not that large, tunning the hyper-parameters a little bit might work out. Please keep me posted if you still cannot reproduce.

**Comment by sheep-for** [comment]
Ok. Thanks again for your patient reply.
